model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala

```diff
@@ -573,7 +573,13 @@ object WorkspaceState {
   case object DeleteFailed extends WorkspaceState
 }
 
-case class WorkspaceSetting(settingType: WorkspaceSettingType, config: WorkspaceSettingConfig)
+sealed abstract class WorkspaceSetting(val settingType: WorkspaceSettingType, val config: WorkspaceSettingConfig)
+
+case class GcpBucketLifecycleSetting(override val config: GcpBucketLifecycleConfig)
+    extends WorkspaceSetting(settingType = WorkspaceSettingTypes.GcpBucketLifecycle, config)
+
+case class GcpBucketSoftDeleteSetting(override val config: GcpBucketSoftDeleteConfig)
+    extends WorkspaceSetting(settingType = WorkspaceSettingTypes.GcpBucketSoftDelete, config)
```

Comment:
I made this change because previously, the way the types were set up allowed for a `WorkspaceSetting` with a `settingType` of `GcpBucketLifecycle` and a `config` of `GcpBucketSoftDeleteConfig` or vice versa. This wasn't reflective of how we want the types to match up. The new sealed class with two subclasses means that any `WorkspaceSetting` will have to either deal with lifecycles or soft delete.